### DISCUSSION TITLE
Naive question on SVGP: are the inducing points `train_x[:500, :]` being re-positioned during training?

### DISCUSSION THREAD

**Discussion by yw5aj** [cornellius-gp/gpytorch#2270] [Q&A]
Hi all,

In the example [here](https://docs.gpytorch.ai/en/stable/examples/04_Variational_and_Approximate_GPs/SVGP_Regression_CUDA.html), there are two lines specifying the inducing points:

```
inducing_points = train_x[:500, :]
model = GPModel(inducing_points=inducing_points)
```

In Hensman's paper [here](https://arxiv.org/abs/1411.2005) they mentioned that "... there are a fixed number of inducing variables...". So should I assume that we are just setting the number of inducing points to be 500 in the code, and the exact points will be optimized during training?

Best,
Shawn

**Comment by gpleiss** [comment]
That is correct. We are fixing the number of inducing points to 500 (standard practice), and optimizing their locations during training.

**Reply by gpleiss** [reply]
I would also mention that (with the update in #2303) we are not repositioning the training data itself: instead we are initializing the inducing points to be the same as training data, and then we are optimizing the locations of these inducing points. So the training data is unaffected.
